core-skills/web-application-development-with-react/README.md

```diff
@@ -0,0 +1,72 @@
+# Web Application Development with React
+
+## Summary
+
+1. Can describe situations where you should/shouldn't decide to use React for a project
+2. Can setup a React project from scratch in a repeatable way
+3. Can describe the different parts of the Javascript ecosystem (e.g. NPM, Webpack, etc)
+4. Can demonstrate using TDD on business logic within a React code base
+5. Can demonstrate strategies to achieve clear separation of business logic and display logic
+with React components
+6. Can demonstrate applying sensible styling to React components
+7. Can demonstrate strategies and tools to run tests against an API within a React application
+
+## Sections
+
+1. [React - Why and when should I use it?](./why-and-when.md)
+2. [Setting up a React project](./setup.md)
+3. [React Fundamentals](./fundamentals/README.md)
+4. [Prototyping React Components](./prototyping.md)
+5. [Styling React Components](./prototyping.md)
+
+## Mark Scheme
+
+### Hedgehog
+
+Learning the fundamentals - How not to hurt yourself with React
+
+- [ ] Can demonstrate an understanding of the benefits of using React
+- [ ] Creates an application with the following:
+  - [ ] The application can be run on any machine (e.g. Using Docker)
+  - Setup/Tooling
+    - The candidate has setup the project with the following tools:
+      - [ ] A sandbox environment to quickly prototype components within (e.g. Storybook)
+      - [ ] Hot-reloading
+      - [ ] Tests (The ability to write them, writing one is not a requirement at this stage)
+  - Components
+    - The candidate is able to explain what the following are:
+      - [ ] A component
+      - [ ] The use of props within components
+      - [ ] The use of state within components
+      - [ ] The use of children within components
+      - [ ] Responding to user events
+- Demonstrating understanding of React fundamentals
+  - The candidate can explain the following:
+      - [ ] How to style components
+      - [ ] The lifecycle of components
+      - [ ] Managing application state
+
+### Fox
+- Components
+  - The candidate has created components demonstrating the following:
+    - [ ] A component without any props or state
+    - [ ] The use of props
+    - [ ] The use of state
+    - [ ] The use of children
+    - [ ] Responding to user events
+    - [ ] Styling components  
+- Architecture
+  - The candidate can explain the following;
+    - [ ] Communicating with APIs
+    - [ ] Division of business logic, domain, rendering and state
+    - [ ] How to test component rendering
+    - [ ] How to test business logic
+
+**Coming Soon - Fox is intended to test ability to write simple components and understanding
```

Comment:
I'd also question the lack of cohesion between understanding architecture and the ability to demonstrate it.  In my eyes, it'd feel more natural to explain why alongside implementing,but I'd like to hear your reasoning.

Reply:
For me, quite a lot of the _how_ is still different from project to project - we're still evolving our process as we go. Knowing we want to split business logic from rendering and why, and how we test this, comes before having actually done it potentially.

Reply:
Yeah, I think `Knowing we want to split business logic from rendering` is a general knowledge thing. It's driven from knowing this is a pattern that applies to whatever tooling you use.

Pulling it off in react is a specific skill that requires putting your knowledge into practice.

Reply:
Are we confident we have a definitive answer on how we do that currently?

Reply:
I wouldn't say we're looking for a definitive answer, but I think we have a few approaches that can work

Reply:
I think we should definitely nail this down when we come to writing out the subsequent parts. I think it's important to try to make sure the person understands what a good React app would be architected like, even if they're not quite sure how to achieve it. Whether or not we decide to split up the theory/application or not is probably good to discuss in person

Reply:
I think the discussion has diverged from both the Hedgehog focus and Adrian's initial question about splitting fox and owl into a semi theoretical component. Is there any problem with Hedgehog being mostly theoretical with some slant toward setup/initialisation?

Reply:
"The candidate can demonstrate *one* method of separating business logic and rendering in a SPA?"

Reply:
I see a huge problem with theoretical - these core skills need to be grounded in Practical skills.

Reply:
Is this in regards to Hedgehog or further levels?

Reply:
I mean, The last level would definitely be "Oh yeah you've actually proven all these things practically". 

With hedgehog I think after the run-through with Steve today we can see whether or not it would make sense to put the implementing fundamentals into this level also, I could see this being a fairly light level otherwise and the others getting too much stuff in them - but we won't know until we run it

Reply:
Core skills should be grounded in practice at all levels, if they are not then this is a smell.

Reply:
If the aim is to prove you're a seasoned react dev, yes. If the aim is to get people to the level where they can contribute to React projects?

Reply:
That said, I'm coming around to the idea of Fox being fairly heavily loaded with practical examples, I agree that examples of components, state management and styling _with appropriate tests_ would make sense, then Owl could have room for more interesting things like business logic, APIs etc

Reply:
I think by the time you've earned the top badge, then you should be setup to become a seasoned react dev in a relatively short period of time e.g. 6-12 months of practice

Reply:
We're on the same page with this I think, it's just harder to tell with Fox & Owl being incomplete.

Reply:
Yeah I think we're all on the same page, should clear up once we have fox/owl more solid. Hopefully within the next few weeks